go/enclave/mempool/manager.go

```diff
@@ -61,118 +66,33 @@ func (db *mempoolManager) FetchMempoolTxs() []*common.L2Tx {
-			return core.ToMap(currentBatch.Transactions), nil
-		}
-
-		if currentBatch.Header.Number.Uint64() == common.L2GenesisHeight {
-			// There's less than `HeightCommittedBlocks` rollups, so there's no transactions to remove yet.
-			return map[gethcommon.Hash]gethcommon.Hash{}, nil
-		}
-
-		currentBatch, err = resolver.FetchBatch(currentBatch.Header.ParentHash)
-		if err != nil {
-			if errors.Is(err, errutil.ErrNotFound) {
-				return nil, fmt.Errorf("found a gap in the rollup chain")
-			}
-			return nil, fmt.Errorf("could not retrieve parent rollup. Cause: %w", err)
-		}
-
-		blocksDeep++
-	}
-}
-
 // CurrentTxs - Calculate transactions to be included in the current batch
-func (db *mempoolManager) CurrentTxs(head *core.Batch, resolver db.BatchResolver) ([]*common.L2Tx, error) {
-	txs, err := findTxsNotIncluded(head, db.FetchMempoolTxs(), resolver)
-	if err != nil {
-		return nil, err
-	}
-	sort.Sort(sortByNonce(txs))
-	return txs, nil
-}
-
-// findTxsNotIncluded - given a list of transactions, it keeps only the ones that were not included in the block
-// todo (#1491) - inefficient
-func findTxsNotIncluded(head *core.Batch, txs []*common.L2Tx, s db.BatchResolver) ([]*common.L2Tx, error) {
-	// go back only HeightCommittedBlocks blocks to accumulate transactions to be diffed against the mempool
-	startAt := uint64(0)
-	if head.NumberU64() > common.HeightCommittedBlocks {
-		startAt = head.NumberU64() - common.HeightCommittedBlocks
-	}
-	included, err := allIncludedTransactions(head, s, startAt)
-	if err != nil {
-		return nil, err
-	}
-	return filterOutTransactions(txs, included), nil
-}
-
-// Recursively finds all transactions included in the past stopAtHeight batches.
-func allIncludedTransactions(batch *core.Batch, s db.BatchResolver, stopAtHeight uint64) (map[gethcommon.Hash]*common.L2Tx, error) {
-	if batch.Header.Number.Uint64() == stopAtHeight {
-		return core.MakeMap(batch.Transactions), nil
-	}
-
-	// We add this batch's transactions to the included transactions.
-	newMap := make(map[gethcommon.Hash]*common.L2Tx)
-	for _, tx := range batch.Transactions {
-		newMap[tx.Hash()] = tx
-	}
-
-	// If the batch has a parent (i.e. it is not the genesis block), we recurse.
-	parentBatch, err := s.FetchBatch(batch.Header.ParentHash)
-	if err != nil && !errors.Is(err, errutil.ErrNotFound) {
-		return nil, err
-	}
-	if err == nil {
-		txsMap, err := allIncludedTransactions(parentBatch, s, stopAtHeight)
-		if err != nil {
-			return nil, err
-		}
-		for hash, tx := range txsMap {
-			newMap[hash] = tx
+func (db *mempoolManager) CurrentTxs(stateDB *state.StateDB) ([]*common.L2Tx, error) {
+	txes := db.FetchMempoolTxs()
+	sort.Sort(sortByNonce(txes))
+
+	applicableTransactions := make(common.L2Transactions, 0)
+	nonceTracker := NewNonceTracker(stateDB)
+
+	for _, tx := range txes {
+		sender, _ := core.GetAuthenticatedSender(db.obscuroChainID, tx)
+		if tx.Nonce() == nonceTracker.GetNonce(*sender) {
```

Comment:
would be good to double check the sender is not nil
Otherwise this whole thing would crash